test/storybook-playwright/specs/button.spec.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { expect, test } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import {
+	gotoStoryId,
+	getAllPropsPermutations,
+	testSnapshotForPropsConfig,
+} from '../utils';
+
+test.describe( 'Button', () => {
+	test.describe( 'variant states', () => {
+		test.beforeEach( async ( { page } ) => {
+			gotoStoryId( page, 'components-button--variant-states', {
+				decorators: { customE2EControls: 'show' },
+			} );
+		} );
+
+		getAllPropsPermutations( [
+			{
+				propName: '__next40pxDefaultSize',
+				valuesToTest: [ true, false ],
+			},
+		] ).forEach( ( propsConfig ) => {
+			test( `should render with ${ JSON.stringify(
+				propsConfig
+			) }`, async ( { page } ) => {
+				await testSnapshotForPropsConfig( page, propsConfig );
+			} );
+		} );
+	} );
+
+	test.describe( 'icon', () => {
+		test.beforeEach( async ( { page } ) => {
+			gotoStoryId( page, 'components-button--icon', {
+				decorators: { customE2EControls: 'show' },
+			} );
+		} );
+
+		getAllPropsPermutations( [
+			{
+				propName: '__next40pxDefaultSize',
+				valuesToTest: [ true, false ],
+			},
+		] ).forEach( ( propsConfig ) => {
+			test( `should render with ${ JSON.stringify(
+				propsConfig
+			) }`, async ( { page } ) => {
+				await testSnapshotForPropsConfig( page, propsConfig );
```

Comment:
Good question. In my testing, it was unnecessary when screenshotting with the `testSnapshotForPropsConfig()` util, but it is necessary otherwise.

I think we might want to try baking the wait into the `gotoStoryId()` util, in combination with a decorator. So `gotoStoryId()` would automatically `waitForSelector( '[data-testid="screenshot-container"]' )` or something. This way we wouldn't have to rely on component classnames. Having a `screenshot-container` wrapper may also be useful for excluding the `customE2EControls` stuff from the screenshots.